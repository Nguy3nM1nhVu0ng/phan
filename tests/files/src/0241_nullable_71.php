<?php
function f0(): string {}
function f1(): ?string {
    return 42;
}
$v = f1();
function f2(): ?string {
    return null;
}
$v = f1();
function f3(?string $name) {}
f3(null);
f3(42);
f3('string');

/**
 * @param ?int $p
 * @return ?string
 */
function f4($p) { return null; }

/**
 * @param int $p
 * @return ?string
 */
function f5($p) { return 42; }

/** @return ?string */
function f6() { return 'string'; }

/** @return ?string[] */
function f7() { return ['string']; }

/** @return ?string[] */
function f8() { return []; }

/** @return ?string[] */
function f9() { return null; }

/** @return Tuple<int,string> */
function f9() { return null; }

