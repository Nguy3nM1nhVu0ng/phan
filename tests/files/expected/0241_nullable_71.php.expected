%s:2 PhanTypeMissingReturn Method \f0 is declared to return string but has no return value
%s:4 PhanTypeMismatchReturn Returning type int but f1() is declared to return ?string
%s:13 PhanTypeMismatchArgument Argument 1 (name) is int but \f3() takes ?string defined at %s:11
%s:26 PhanTypeMismatchReturn Returning type int but f5() is declared to return ?string
%s:32 PhanTypeMismatchReturn Returning type string[] but f7() is declared to return ?string
%s:35 PhanTypeMismatchReturn Returning type array but f8() is declared to return ?string
%s:41 PhanRedefineFunction Function f9 defined at nullable.php:41 was previously defined at %s:38
